packages/core/src/services/evaluationResults/create.test.ts

```diff
@@ -199,8 +204,33 @@ describe('createEvaluationResult', () => {
     })
 
     expect(result.ok).toBe(false)
-    if (result.error) {
-      expect(result.error.message).toContain('Unsupported result type')
-    }
+    expect(result.error!.message).toContain('Unsupported result type')
+  })
+
+  it('saves unsupported type if result is not defined', async () => {
```

Comment:
If we have an evaluation_result without `result` it means something bad happend. Between the things that could happen is that we added for some reason a new evaluation type that's not valid in the DB. In that case we store the error and the evaluation result with `resultableType: null`